### ISSUE TITLE
Regression for assume role example

### ISSUE THREAD

**Issue by charignon** [cognitect-labs/aws-api#130]
Thank you for your interest in helping to improve Cognitect's aws-api!
## Dependencies

When I run the example for [assume role](https://github.com/cognitect-labs/aws-api/blob/master/examples/assume_role_example.clj) with the following set of dependencies:

``` clojure
{:deps {com.cognitect.aws/api       {:mvn/version "0.8.408"}
        com.cognitect.aws/endpoints {:mvn/version "1.1.11.686"}
        com.cognitect.aws/kms       {:mvn/version "791.2.615.0"}
        com.cognitect.aws/sts       {:mvn/version "773.2.578.0"}
        com.cognitect.aws/iam       {:mvn/version "788.2.609.0"}}}
```

it works fine

But when I use the more recent dependencies
```clojure
{:deps {com.cognitect.aws/api       {:mvn/version "0.8.437"}
        com.cognitect.aws/endpoints {:mvn/version "1.1.11.726"}
        com.cognitect.aws/kms       {:mvn/version "791.2.615.0"}
        com.cognitect.aws/sts       {:mvn/version "773.2.578.0"}
        com.cognitect.aws/iam       {:mvn/version "788.2.609.0"}}}
```

It hangs seemingly forever (I let it run a few minutes) on:
```
(aws/invoke iam-with-assumed-role {:op :GetUser :request {:UserName (:UserName me)}})
```

The code example is the one provided in the repo that I linked above.

## Stack traces
No stack trace

@dchelimsky FYI

**Comment by charignon** [comment]
All the other aspects of the demo seem to be working fine except this.

**Comment by dchelimsky** [comment]
The problem is that we've got a shared threadpool of 1 for credentials providers here: https://github.com/cognitect-labs/aws-api/blob/2d57c12218ed4cdc1bc8584ab2a1ecf689aff3f3/src/cognitect/aws/credentials.clj#L42. The `assumed-role-credentials-provider` issues its request first, which triggers a request to the inner `sts`'s client, except that one queues up, waiting for the first one to return before the pool will give it a thread to work with.

I'm working on a workaround (until we fix it properly) and will post it here.

**Comment by dchelimsky** [comment]
I couldn't come up with a good workaround, so I just fixed it in release 0.8.445.
